src/gcp/cloudsql/permissions.ts

```diff
@@ -221,22 +276,316 @@ export async function setupSQLPermissions(
     revokes.push(`REVOKE "cloudsqlsuperuser" FROM "${firebaseowner(databaseId)}"`);
   }
 
+  const user = (await getIAMUser(options)).user;
+  const projectNumber = await needProjectNumber(options);
+  const { user: fdcP4SAUser } = toDatabaseUser(getDataConnectP4SA(projectNumber));
+
   const sqlRoleSetupCmds = concat(
     // For backward compatibality we sometimes need to revoke some roles.
     revokes,
 
     // We shoud make sure schema exists since this setup runs prior to executing the diffs.
-    [`CREATE SCHEMA IF NOT EXISTS "public"`],
+    [`CREATE SCHEMA IF NOT EXISTS "${schema}"`],
```

Comment:
In the backend code, I moved this to [`ownerRolePermissions`](https://source.corp.google.com/piper///depot/google3/firebase/dataconnect/shared/sqlpermission/permissions.go;l=53?q=f:permission&ss=piper%2FGoogle%2FPiper:google3%2Ffirebase%2Fdataconnect%2F) because there is a chance for the schema to be created with `firebasesuperuser` as the owner, and then switch to `firebaseowner`.